src/main/java/org/reaktivity/nukleus/oauth/internal/OAuthController.java

```diff
@@ -105,21 +101,18 @@ public String name()
         return controllerSpi.doResolve(resolveRO.typeId(), resolveRO.buffer(), resolveRO.offset(), resolveRO.sizeof());
     }
 
-    // TODO: be able to get the extension to be read by doResolve(). does not seem to read extensions
-    //       extensions can be added successfully to the flyweights, BUT are not read by doResolve().
     public CompletableFuture<Long> resolve(
         String realmName,
         String issuerName,
         String audienceName,
         String[] roles)
     {
         long correlationId = controllerSpi.nextCorrelationId();
-
-        OAuthResolveExFW resolveEx = resolveExRW.wrap(commandBuffer, 0, commandBuffer.capacity())
+        final UnsafeBuffer resolveExBuffer = newUnsafeBuffer();
```

Comment:
Rename `resolveExBuffer` to `extensionBuffer` and avoid allocating on each call to `resolve` by hoisting to a `private final` member field alongside `commandBuffer`.

Inline the assignments to `commandBuffer` and `extensionBuffer` in constructor, no need for separate method here.